eva_cttv_pipeline/evidence_string_generation/annotated_clinvar.py

```diff
@@ -0,0 +1,85 @@
+import xml.etree.ElementTree as ET
+
+from eva_cttv_pipeline.clinvar_xml_io.clinvar_xml_io import ClinVarTrait, ClinVarRecordMeasure, ClinVarDataset, \
+    ClinVarRecord
+from eva_cttv_pipeline.clinvar_xml_io.clinvar_xml_io.xml_parsing import iterate_rcv_from_xml
+from eva_cttv_pipeline.evidence_string_generation.clinvar_to_evidence_strings import load_efo_mapping, \
+    get_consequence_types
+from eva_cttv_pipeline.evidence_string_generation import consequence_type as CT
+
+
+PROVIDER = 'ClinVarXmlPipeline'
+
+
+class AnnotatingClinVarDataset(ClinVarDataset):
+    """This class provides the ability to annotate ClinVar records with EFO mappings and consequence mappings."""
+
+    def __init__(self, clinvar_xml, string_to_efo_mappings, variant_to_gene_mappings):
+        super().__init__(clinvar_xml)
+        self.header = f'''<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<ReleaseSet Dated="{self.today()}" ProcessedBy="{PROVIDER}" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Type="full" xsi:noNamespaceSchemaLocation="http://ftp.ncbi.nlm.nih.gov/pub/clinvar/xsd_public/clinvar_public_1.60.xsd">
```

Comment:
Actually, I suppose we're not really following ClinVar's XML schema any more so might need to change that value anyway (and I guess publish our own modified xsd?  I'm not too familiar with how this works).

And yes you're right, we should probably preserve at least some bits of the header of the original file, e.g. the date (which isn't currently parsed but would actually be useful for us running the pipeline for OT anyway).

Reply:
I believe that, the closest we could get to ClinVar's output XML, the better. If we could annotate the XML within the boundaries of their schema, even better. I say this because it would be way easier to streamline into any other process that uses ClinVar's XMLs.

Regarding the headers: other files, like CRAMs, keep track of every process that the file suffered, every transformation, within the headers. In my personal experience this is a very welcomed feature, for traceability of changes. If there is the possibility of keeping the original header, or part of it, but replace the important/changed bits with ours, the better

Reply:
I've now updated the code to preserve original header attributes and add extra attributes for our processing, see [updated diff](https://www.diffchecker.com/b7u9Y576).  It's not quite the full traceability of CRAMs or VCFs (in particular it loses information if you run multiple times) but I think having that would be a fairly complicated change... e.g. things would have to be stored in new elements instead of attributes, etc.  Let me know what you think.